services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -47,31 +51,22 @@ class HankeController(@Autowired private val hankeService: HankeService) {
 
     @GetMapping
     fun getHankeList(hankeSearch: HankeSearch? = null): ResponseEntity<Any> {
-
-
-        if (hankeSearch != null && hankeSearch.saveType != null) {
-            return try {
-                val hankeList = hankeService.loadAllHankeWithSavetype(hankeSearch.saveType)
-                ResponseEntity.status(HttpStatus.OK).body(hankeList)
-
-            } catch (e: Exception) {
-                logger.error(e) {
-                    HankeError.HAI1004.toString()
-                }
-                ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(HankeError.HAI1004)
-            }
-        }
-
-        if (hankeSearch == null || hankeSearch.periodBegin == null || hankeSearch.periodEnd == null ) {
-            return getAllHankeItems()
-        }
         return try {
-            //  Get all hanke datas within time period (= either or both of alkuPvm and loppuPvm are inside the requested period)
-            // TODO: user token  from front?
-            //  TODO: do we limit result for user own hanke?
-            val hankeList = hankeService.loadAllHankeBetweenDates(hankeSearch.periodBegin!!, hankeSearch.periodEnd!!)
+            // TODO it might be better to move this logic with different search types into service and keep controller as clean and lean as possible
```

Comment:
That is not bad idea. Should have thought that when controller started to get more complex.